includes/class-edd-customer.php

```diff
@@ -660,40 +660,43 @@ public function get_payments( $status = array() ) {
 	 *
 	 * @since 2.3
 	 *
-	 * @param int  $payment_id   The payment ID to attach to the customer.
+	 * @param int  $order_id     The Order ID to attach to the customer.
 	 * @param bool $update_stats For backwards compatibility, if we should increase the stats or not.
 	 *
 	 * @return bool True if the attachment was successfully, false otherwise.
 	 */
-	public function attach_payment( $payment_id = 0, $update_stats = true ) {
+	public function attach_payment( $order_id = 0, $update_stats = true ) {
 
-		// Bail if no payment ID
-		if ( empty( $payment_id ) ) {
+		// Bail if no payment ID.
+		if ( empty( $order_id ) ) {
 			return false;
 		}
 
-		// Get payment
-		$order = edd_get_order( $payment_id );
+		// Get order.
+		$order = edd_get_order( $order_id );
 
-		// Bail if payment does not exist
+		// Bail if payment does not exist.
 		if ( empty( $order ) ) {
 			return false;
 		}
 
-		// Bail if already attached
-		if ( (int) $order->customer_id === (int) $this->id ) {
-			return true;
-		}
-
 		do_action( 'edd_customer_pre_attach_payment', $order->id, $this->id, $this );
 
-		// Update the order
-		$success = (bool) edd_update_order( $payment_id, array(
-			'customer_id' => $this->id,
-			'email'       => $this->email
-		) );
+		$success = (int) $order->customer_id === (int) $this->id;
+
+		// Update the order if it isn't already attached.
+		if ( ! $success ) {
+			// Update the order.
+			$success = (bool) edd_update_order(
+				$order_id,
+				array(
+					'customer_id' => $this->id,
+					'email'       => $this->email,
```

Comment:
@robincornett Yeah I had that same thought, but I think since we were already doing this, it's safer to leave this as is, and we can further look at if this has any unintended side affects at a later date.

With this changeset we aren't changing anything from our previous behavior of updating the email address.